pkg/drivers/vfkit/vfkit.go

```diff
@@ -370,6 +370,10 @@ func (d *Driver) setupIP(mac string) error {
 		if err == nil {
 			break
 		}
+		// When DHCP discovery is slow or stuck, surface vfkit state to aid debugging.
+		if i > 0 && i%10 == 0 { // every 10th attempt
+			d.logVFKitStatus(fmt.Sprintf("ip lookup still failing (attempt %d)", i))
+		}
```

Comment:
Instead of logging, we should call d.GetState() and exit with an error if we are not in running state. This should be done on every iteration. There is no point in running 20 seconds if vfkit terminated.